File: ios/Models/ProcessorAdditionalSessionData.swift
import Foundation

public class ProcessorAdditionalSessionData: JSONParams {
  private var object: [AnyHashable: Any?]

  override init(target: [AnyHashable: Any]?) {
    self.object = [:]
    
    super.init(target: target)
    
    self.parseJSON()
  }

  private func parseJSON() {
    let platform = self.getParam("platform", type: String.self)
    let appID = self.getParam("appID", type: String.self)
    let installationID = self.getParam("installationID", type: String.self)
    let deviceModel = self.getParam("deviceModel", type: String.self)
    let deviceSDKVersion = self.getParam("deviceSDKVersion", type: String.self)
    let userAgent = self.getParam("userAgent", type: String.self)
    let sessionID = self.getParam("sessionID", type: String.self)

    self.object["platform"] = platform
    self.object["appID"] = appID
    self.object["installationID"] = installationID
    self.object["deviceModel"] = deviceModel
    self.object["deviceSDKVersion"] = deviceSDKVersion
    self.object["userAgent"] = userAgent
    self.object["sessionID"] = sessionID
  }

  func getDict() -> NSDictionary {
    return NSDictionary(dictionary: self.object as [AnyHashable : Any])
  }
}

File: android/src/main/java/com/azifacemobile/models/ProcessorAdditionalSessionData.java
package com.azifacemobile.models;

import com.azifacemobile.models.abstracts.CommonParams;
import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.WritableMap;

import org.json.JSONObject;

public class ProcessorAdditionalSessionData extends CommonParams {
  private final WritableMap object;

  public ProcessorAdditionalSessionData(JSONObject target) {
    super(target);

    this.object = Arguments.createMap();

    this.parseJSON();
  }

  private void parseJSON() {
    String platform = this.getParam( "platform", String.class);
    String appID = this.getParam("appID", String.class);
    String installationID = this.getParam("installationID", String.class);
    String deviceModel = this.getParam("deviceModel", String.class);
    String deviceSDKVersion = this.getParam("deviceSDKVersion", String.class);
    String userAgent = this.getParam("userAgent", String.class);
    String sessionID = this.getParam("sessionID", String.class);

    this.object.putString("platform", platform);
    this.object.putString("appID", appID);
    this.object.putString("installationID", installationID);
    this.object.putString("deviceModel", deviceModel);
    this.object.putString("deviceSDKVersion", deviceSDKVersion);
    this.object.putString("userAgent", userAgent);
    this.object.putString("sessionID", sessionID);
  }

  public WritableMap getMap() {
    return this.object.copy();
  }
}
